contracts/cw-pre-propose-base/src/msg.rs

```diff
@@ -0,0 +1,55 @@
+use schemars::JsonSchema;
+use serde::{Deserialize, Serialize};
+
+use voting::deposit::UncheckedDepositInfo;
+
+#[derive(Serialize, Deserialize, Clone, Debug, PartialEq, JsonSchema)]
+#[serde(rename_all = "snake_case")]
+pub struct InstantiateMsg<InstantiateExt> {
+    /// Information about the deposit requirements for this
+    /// module. None if no deposit.
+    pub deposit_info: Option<UncheckedDepositInfo>,
+    /// If false, only members (addresses with voting power) may create
+    /// proposals in the DAO. Otherwise, any address may create a
+    /// proposal so long as they pay the deposit.
+    pub open_proposal_submission: bool,
+    /// Extension for instantiation. The default implementation will
+    /// do nothing with this data.
+    pub ext: InstantiateExt,
+}
+
+#[derive(Serialize, Deserialize, Clone, Debug, PartialEq, JsonSchema)]
+#[serde(rename_all = "snake_case")]
+pub enum ExecuteMsg<ProposalMessage, ExecuteExt> {
+    /// Creates a new proposal in the pre-propose module. MSG will be
+    /// serialized and used as the proposal creation message.
+    Propose { msg: ProposalMessage },
+
+    /// Extension message. Contracts that extend this one should put
+    /// their custom execute logic here. The default implementation
+    /// will do nothing if this variant is executed.
+    Ext { msg: ExecuteExt },
```

Comment:
CosmWasm should develop some conventions for this pattern. In `cw-nfts` we always call it `Extension`.

Reply:
Great point. Verbose is good. I'll update this to use `Extension`.